src/coverlet.msbuild.tasks/CoverageResultTask.cs

```diff
@@ -75,13 +75,14 @@ public override bool Execute()
             {
                 Console.WriteLine("\nCalculating coverage result...");
 
-                if (InstrumenterState is null || !File.Exists(InstrumenterState.ItemSpec))
+                IFileSystem fileSystem = (IFileSystem) DependencyInjection.Current.GetService(typeof(IFileSystem));
+                if (InstrumenterState is null || !fileSystem.Exists(InstrumenterState.ItemSpec))
                 {
                     _logger.LogError("Result of instrumentation task not found");
                     return false;
                 }
 
-                var coverage = new Coverage(CoveragePrepareResult.Deserialize(new FileStream(InstrumenterState.ItemSpec, FileMode.Open)), this._logger, (IInstrumentationHelper)DependencyInjection.Current.GetService(typeof(IInstrumentationHelper)));
+                var coverage = new Coverage(CoveragePrepareResult.Deserialize(new FileStream(InstrumenterState.ItemSpec, FileMode.Open)), this._logger, (IInstrumentationHelper)DependencyInjection.Current.GetService(typeof(IInstrumentationHelper)), fileSystem);
```

Comment:
Why not also for `new FileStream(InstrumenterState.ItemSpec, FileMode.Open)`?
And we should surround by a `using` to close handle asap(my past error).

Reply:
Done